### ISSUE TITLE
Support profile inheritance on optional profiles

### ISSUE THREAD

**Issue by agjohnson** [PyCQA/prospector#161]
It seems that configuration inheritance ensures the profile exists. Would inheritance for optional files be a useful feature, and if so, what would the best implementation of this be? I'm happy to get this working if it would be an accepted feature.

My use case is running prospector using vim, and having projects with a base configuration file. I'd like to optionally include a local override profile, but only if it exists. The local override profile wouldn't be checked into the project, but would provide a method for stricter linting for individual developers. For example, consider the following `prospector.yaml`:

``` yaml
inherits:
  - prospector_local
  - prospector_base
```

Where `prospector_base.yml` would be the base project configuration, and `prospector_local.yml` wouldn't necessarily exist in the repository. A couple implementation options here:
- Add to the config spec to signify optional profiles
- Add a configuration option that flags missing profiles as soft errors rather than hard. Default is the normal behavior, present now. To silently skip a missing profile would be unexpected user experience.

It's also possible I'm missing something and there is an easier method to accomplish what I'm talking about.

**Comment by carlio** [comment]
I think I'd quite like to do this for no particular reason other than I thought of a nice idea.

```
inherits:
  - prospector_local?
  - prospector_base
```

Sort of like Typescript to mark it as optional. I think I'll do this just cos I think it'd be cool :-)

**Comment by carlio** [comment]
Only took 6 years but it's finally implemented in 1.7.5 :partying_face:
